fix: add secure context checks for navigator.mediaDevices

yrendaoai opened the pull request:
# Fix: Support LiveKit Client in Non-Secure Context (HTTP Environment)

##  Problem

The original code directly calls `navigator.mediaDevices` APIs without checking for secure context. In HTTP environments (non-HTTPS), these APIs are restricted by browser security policies:

- `navigator.mediaDevices` may be `undefined`
- `addEventListener('devicechange')` fails even if `mediaDevices` exists
- `getUserMedia()` calls throw errors

This causes the WebRTC video player to fail in HTTP environments, breaking the live video streaming functionality.

##  Solution

- **Added secure context checks** before accessing `navigator.mediaDevices` APIs

##  Changes

### 1. Code Changes in Fixed PR
The fixed PR adds security context checks in `src/room/Room.ts`:

**Before:**
```typescript
navigator.mediaDevices?.addEventListener('devicechange', this.handleDeviceChange, {
  signal: abortController.signal,
});
```

**After:**
```typescript
if (window.isSecureContext && navigator.mediaDevices?.addEventListener) {
  navigator.mediaDevices.addEventListener('devicechange', this.handleDeviceChange, {
    signal: abortController.signal,
  });
}
```

Fixed locations:
-  Device change event listener registration (constructor)
-  Device change event listener removal (cleanup)
-  `getUserMedia()` calls for video tracks
-  `getUserMedia()` calls for audio tracks

##  Testing

-  Tested in HTTP environment - no errors when accessing `navigator.mediaDevices`
-  Tested in HTTPS environment - functionality works as expected
-  WebRTC video player works correctly in both secure and non-secure contexts
-  Device change events are properly handled when available

##  Related

This fix ensures compatibility with HTTP environments while maintaining full functionality in HTTPS environments. The changes are backward compatible and do not affect existing HTTPS deployments.

##  Notes

- The fixed PR maintains the same API as the original `livekit-client`
- All features work identically in secure contexts
- In non-secure contexts, device change events are gracefully skipped
- `getUserMedia()` calls fall back to dummy tracks when not available

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/livekit/client-sdk-js?pullRequest=1769) <br/>All committers have signed the CLA.

lukasIO reviewed (commented):
thanks for the PR and the detailed description!

yrendaoai commented on src/room/Room.ts:
```diff
@@ -2499,8 +2499,7 @@ class Room extends (EventEmitter as new () => TypedEmitter<RoomEventCallbacks>)
-            ? (
-                await window.navigator.mediaDevices.getUserMedia({ video: true })
+            ? (await window.navigator.mediaDevices?.getUserMedia?.({ video: true }) ?? new MediaStream()
               ).getVideoTracks()[0]
```
Note: The `getUserMedia` calls with `?? new MediaStream()` fallback (lines 2502, 2529) 
ensure we don't get undefined when calling `.getVideoTracks()[0]` in non-secure contexts 
where `mediaDevices?.getUserMedia?.()` returns undefined due to optional chaining.

yrendaoai commented:
@lukasIO Thanks for the feedback! I've updated the PR to use optional chaining as suggested and moved the condition checks upfront for getUserMedia calls. Please take a look when you have a chance.
